sklearn/tree/_criterion.pxd

```diff
@@ -40,6 +40,14 @@ cdef class Criterion:
     cdef double weighted_n_left          # Weighted number of samples in the left node
     cdef double weighted_n_right         # Weighted number of samples in the right node
 
+    cdef double* node_sum_total          # An array of the current node value
+    cdef double* node_sum_left           # An array of the left node value
+    cdef double* node_sum_right          # An array of the right node value
```

Comment:
They are not different, it is just a renaming. It's to share them between classification and regression, so that when we do caching between splits we have a single value to pass through the splitter regardless of criteria.